docs/source/cpp/orc.rst

```diff
@@ -0,0 +1,126 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+
+.. cpp:namespace:: arrow::adapters::orc
+
+=============================
+Reading and Writing ORC files
+=============================
+
+The `ORC format <https://orc.apache.org/docs/>`__
+is a space-efficient columnar storage format for complex data. 
+Arrow provides a fast ORC reader and a fast ORC writer allowing
+ingestion of external data as Arrow tables.
+
+.. seealso::
+   :ref:`ORC reader/writer API reference <cpp-api-orc>`.
+
+Supported ORC features
+==========================
+
+The ORC format has many features, and we support a subset of them.
+
+Compression
+-----------
+
++-------------------+---------+
+| Compression codec | Notes   |
```

Comment:
No need for a notes column if there aren't any notes.

Reply:
Now there are notes..